### PR TITLE
Documentation: add upgrade gotchas/further info for better visibility on google, etc

### PR DESCRIPTION
**Pull request by andrewstuart**
This PR updates the documentation with the two root causes of some issues experienced during upgrade, and includes the log output I was receiving to hopefully help users making the upgrade more quickly find the info I eventually found while troubleshooting.

### CODE REVIEW

File: Documentation/upgrades/upgrade_3_0.md [hunk 1]
```diff
@@ -116,4 +116,21 @@ $ ETCDCTL_API=3 etcdctl endpoint health
...
+
+- etcdctl environment variables have been updated. If you find that
+    `ETCDCTL_API=2 etcdctl cluster-health` works properly but `ETCDCTL_API=3
+    etcdctl endpoints health` responds with `Error:  grpc: timed out when
+    dialing`, make sure you are using the [new variable
+    names](https://github.com/coreos/etcd/tree/master/etcdctl#etcdctl).
+
+## Known Issues
+
+- etcd 3.0.x does not work properly on Go 1.7+. See [Issue
```
Comment (gyuho) [inline Documentation/upgrades/upgrade_3_0.md]: This only breaks if etcd v3.0.x is built manually with Go 1.7+, so maybe

> Use Go 1.6.x to build etcd v3.0.x; see [Issue...

**Comment by gyuho** [comment]
Also could you change your commit title to `Documentation: add upgrade gotchas/further info for better visibility on google, etc`?

Thanks!

File: Documentation/upgrades/upgrade_3_0.md (hunk 1 above)
Comment (andrewstuart) [inline Documentation/upgrades/upgrade_3_0.md]: Let me know if the changes I just pushed help clarify the problem and solutions a bit better. :-)

**Comment by gyuho** [comment]
@andrewstuart I meant the actual commit title `Documentation: add upgrade gotchas/further info for better visibility on google, etc`? See https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#format-of-the-commit-message for more contribution guides.

Also could you change and squash commits into one?
Then looks good.

Thanks!

**Comment by andrewstuart** [comment]
Done :-)

**Comment by xiang90** [comment]
@andrewstuart LGTM. Thank you!

**Pull request merged**